ROADMAP.md

```diff
@@ -8,60 +8,57 @@ enhancement process, please visit the enhancements [repository](https://github.c
 
 This document tracks only major changes done to Keylime. 
 
-## Rust Agent
-#### End of Q3-2022
-**Status:** mostly complete
-
-The Keylime agent is being ported to Rust Lang. This decision was made based on
-several reasons:
-
-* Rust is statically linked and does not require the ability to retrieve
-  dependencies. (Important for non internet connected machines or immutable read
-  only operating systems)
-* Rust can be more performant and generally requires less resources.
-* Rust provides strong safety guarantees (memory safe).
-
-For further details of development, please consult the [rust-keylime](https://github.com/keylime/rust-keylime)
-repository and the [Rust agent roadmap issue](https://github.com/keylime/keylime/issues/986).
-
 ## Removal of the Python Agent
-#### End of Q4-2022
+#### Q1-2023
 
 Once the Rust agent is stable the Python agent will be removed in two stages:
 
 1. Add deprecation warning and keep it for one major release
 2. Remove the agent from the code with the release of 7.0.0
 
 ## User Experience Improvements
-#### End of Q4-2022
 
-Some aspects of Keylime of the Keylime user experience can be improved: 
+Some aspects of Keylimes user experience can be improved: 
 
 * Update the user documentation (https://github.com/keylime/keylime/issues/1035)
-* Simplify the TLS setup (https://github.com/keylime/enhancements/pull/73)
-* Simplify the configuration (https://github.com/keylime/enhancements/pull/73)
-* Remove or rework WebUI
 * Investigate integration with monitoring systems (e.g. Prometheus)
 
-## Push Model
-#### Q4-2022 or Q1-2023
 
-Instead of the verifier connecting to the agent to retrieve the attestation data, 
-the agent can also send this data periodically to the verifier.
+## Move Keylime to a Plugin Architecture and Refactoring
+### Q2/Q3-2023
+
+The main eventloop in Keylime is very focused on TPM based attestation in combination with IMA and Measured Boot.
+This has the disadvantage that it is currently not easy to support other forms of claims and evidence 
+(e.g. Intel SGX, AMD SEV) and their retrieval via different methods (e.g. push model, one shot attestation).
+Moving to a more flexible plugin architecture allows us to implement those changes without requiring core changes to Keylime. 
+
+This entails the following aspects:
+* Evaluating the use of general policy languages for validation (e.g. Rego or Seedwing)
+* Use common remote attestation terminology (see where the current one differs to the [rats](https://datatracker.ietf.org/wg/rats/about/) one)
+* Defining a plugin API and runtime specification
+* Moving the current validation parts into separate plugins: quote validation, IMA validation, static PCR checks, Measured Boot
+* Implement the pull model as the default runtime
```

Comment:
Is the  SEV / SGX a push model when the client initiate the communication with the verifier for each event / change, or can work on the same pull model used today?

Reply:
I can attest that it works with the pull model. We have an initial (and **extremely** crude prototype) where we simply inserted an "SEV attestation report evaluator" function into the main eventloop (I didn't write code, and don't even dream of opening a PR on it, but it is a "proof of existence").